tornado/gen.py

```diff
@@ -293,6 +293,10 @@ def _make_coroutine_wrapper(func, replace_callback):
     if hasattr(types, 'coroutine'):
         func = types.coroutine(func)
 
+    stack_context_state = stack_context._state
+    Return_ = Return
+    GeneratorType_ = GeneratorType
```

Comment:
Why are you using this tmp? I think that GeneratorType_ is only used once.

Reply:
Once... *per call*